clients/src/main/java/org/apache/kafka/common/record/FileLogInputStream.java

```diff
@@ -71,7 +71,8 @@ public FileChannelRecordBatch nextBatch() throws IOException {
 
         // V0 has the smallest overhead, stricter checking is done later
         if (size < LegacyRecord.RECORD_OVERHEAD_V0)
-            throw new CorruptRecordException(String.format("Record size is smaller than minimum record overhead (%d).", LegacyRecord.RECORD_OVERHEAD_V0));
+            throw new CorruptRecordException(String.format("Found record size %d smaller than minimum record overhead (%d) in file %s.",
+                    size, LegacyRecord.RECORD_OVERHEAD_V0, channel));
```

Comment:
`toString` of `FileChannel` doesn't seem to be useful:

```text
scala> FileChannel.open(new File("/Users/ijuma/Downloads/trace.log").toPath)
res5: java.nio.channels.FileChannel = sun.nio.ch.FileChannelImpl@591e5fd4
```

Reply:
Yeah, you're right. I looked at the implementation, but saw the `toString` of an inner class by mistake.